cmd/soroban-cli/src/commands/contract/bindings/typescript.rs

```diff
@@ -87,8 +89,15 @@ impl NetworkRunnable for Cmd {
             let wasm: wasm::Args = wasm.into();
             wasm.parse()?.spec
         } else {
-            let contract_id = soroban_spec_tools::utils::contract_id_from_str(&self.contract_id)
-                .map_err(|e| Error::CannotParseContractId(self.contract_id.clone(), e))?;
+            let network = config.map_or_else(
+                || self.network.get(&self.locator).map_err(Error::from),
+                |c| c.get_network().map_err(Error::from),
```

Comment:
```suggestion
            let network = config.map_or_else(
                || Ok(self.network.get(&self.locator)?),
                |c| Ok(c.get_network()?),
```
This does the same thing.

Reply:
With this change I get this:

```
error[E0282]: type annotations needed
  --> cmd/soroban-cli/src/commands/contract/bindings/typescript.rs:93:20
   |
93 |                 || Ok(self.network.get(&self.locator)?),
   |                    ^^ cannot infer type of the type parameter `E` declared on the enum `Result`
   |
help: consider specifying the generic arguments
   |
93 |                 || Ok::<commands::network::Network, E>(self.network.get(&self.locator)?),
   |                      +++++++++++++++++++++++++++++++++

For more information about this error, try `rustc --explain E0282`.
error: could not compile `soroban-cli` (lib) due to previous error
```